kazoo/tests/test_threading_handler.py

```diff
@@ -200,6 +200,28 @@ def wait_for_val():
         eq_(lst, [True])
         th.join()
 
+    def test_wait_race(self):
```

Comment:
> I thought about adding a test, but I didn't do it because a failure would cause the test code to hang forever.

Oh yeah. I forgot about that. 😄 

Your test looks good to me, however it's just complicated enough that it's not immediately apparent what you're trying to test. So do you mind adding a simple docstring that explains _what_ this is trying to test/guard against?

Maybe something like:

    """Test that there is no race condition in `IAsyncResult.wait()`.
         
    Guards against https://github.com/python-zk/kazoo/issues/485 reappearing.
    """"